### PR TITLE
HDX-9819 Adapt the dataset and resource endpoints to the new schema

### PR DESCRIPTION
**Pull request by IanHopkinson**
This PR makes updates to the new HAPI schema for the dataset and resource endpoints.

Note that the HAPI API uses title and dataset_title here, as specified in the spreadsheet which is inconsistent with usage elsewhere hdx_name is used (alongside hdx_stub)

I need to add a get_organizational_api_url function and add the related field to the API.

### CODE REVIEW

**Review by danmihaila — APPROVED** [review]

File: tests/test_endpoints/endpoint_data.py [hunk 1]
```diff
@@ -301,29 +303,30 @@
...
             'update_date_min': date(2023, 6, 1),
-            'update_date_max': date(2023, 6, 2),
+            'update_date_max': date(2023, 6, 3),
             'is_hxl': True,
-            'dataset_hdx_id': 'c3f001fa-b45b-464c-9460-1ca79fd39b40',
+            'hapi_updated_date': date(2023, 6, 2),
+            'dataset_hdx_stub': 'dataset01',
             'dataset_title': 'Dataset #1',
-            'dataset_hdx_provider_stub': 'pRoViDeR01',
+            'dataset_hdx_provider_stub': 'provider01',
```
Comment (alexandru-m-g) [inline tests/test_endpoints/endpoint_data.py:318]: I think this weird looking string, "pRoViDeR01",  was there to tests case insensitivity

File: tests/test_endpoints/endpoint_data.py [hunk 2]
```diff
@@ -301,29 +303,30 @@
             'admin2_name',
             'admin1_ref',
         ],
-    },
-    '/api/resource': {
+    },  # (hdx_id, dataset_hdx_id, name, format, update_date, is_hxl, download_url, hapi_updated_date)
```
Comment (alexandru-m-g) [inline tests/test_endpoints/endpoint_data.py:306]: is the commented line here on purpose ?

**Review by alexandru-m-g — APPROVED** [review]

File: tests/test_endpoints/endpoint_data.py (hunk 1 above)
Comment (IanHopkinson) [reply tests/test_endpoints/endpoint_data.py:318]: I think I must have copied it over from elsewhere, I have no problem with the wierd casing.

File: tests/test_endpoints/endpoint_data.py (hunk 2 above)
Comment (IanHopkinson) [reply tests/test_endpoints/endpoint_data.py:306]: I think it is a remnant of the process I used to make the file, so it can be deleted.

**Pull request merged**